### PR TITLE
fix several erroneous content types

### PR DESCRIPTION
**Pull request by fvanderbiest**
This PR addresses #1304 and also other issues with content-type

### CODE REVIEW

**Comment by fvanderbiest** [comment]
Oops, failing tests ... 
```
-------------------------------------------------------
 T E S T S
-------------------------------------------------------
Running org.georchestra.ldapadmin.ws.backoffice.users.UsersExportTest
Tests run: 4, Failures: 0, Errors: 0, Skipped: 2, Time elapsed: 0.632 sec - in org.georchestra.ldapadmin.ws.backoffice.users.UsersExportTest
Running org.georchestra.ldapadmin.ws.backoffice.users.UsersControllerTest
Tests run: 22, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.622 sec <<< FAILURE! - in org.georchestra.ldapadmin.ws.backoffice.users.UsersControllerTest
testResquestProducesDelete(org.georchestra.ldapadmin.ws.backoffice.users.UsersControllerTest)  Time elapsed: 0.009 sec  <<< FAILURE!
java.lang.AssertionError: null
	at org.junit.Assert.fail(Assert.java:92)
	at org.junit.Assert.assertTrue(Assert.java:43)
	at org.junit.Assert.assertTrue(Assert.java:54)
	at org.georchestra.ldapadmin.ws.backoffice.users.UsersControllerTest.testResquestProducesDelete(UsersControllerTest.java:556)
```

**Comment by Vampouille** [comment]
LGTM !
Maybe we can move all content type declarations in `@RequestMapping` annotations. This must be tested because sometimes it does not work with content type definition in annotation.

File: mapfishapp/src/main/java/org/georchestra/mapfishapp/ws/DocController.java [hunk 1]
```diff
@@ -358,7 +358,7 @@ private void storeFile(A_DocService docService, String docUrl, HttpServletReques
 
             // send back to client the url path to retrieve this file later on
             response.setStatus(HttpServletResponse.SC_CREATED); // 201 created, new resource created
-            response.setContentType("text/html; charset=utf-8");
+            response.setContentType("application/json");
```
Comment (jdenisgiguere) [inline mapfishapp/src/main/java/org/georchestra/mapfishapp/ws/DocController.java]: The change broke AJAX Form request.
From http://docs.sencha.com/extjs/3.4.0/#!/api/Ext.form.BasicForm : 

> The server response is parsed by the browser to create the document for the IFRAME. If the server is using JSON to send the return object, then the Content-Type header must be set to "text/html" in order to tell the browser to insert the text unchanged into the document body.

Comment (fvanderbiest) [inline mapfishapp/src/main/java/org/georchestra/mapfishapp/ws/DocController.java]: Nice catch, Niels !

Comment (fvanderbiest) [inline mapfishapp/src/main/java/org/georchestra/mapfishapp/ws/DocController.java]: Ugly ExtJS here... fixed with 794dc3d

**Pull request merged**